crates/context_aware_config/src/api/custom_types/handlers.rs

```diff
@@ -0,0 +1,160 @@
+        unexpected_error!("could not parse regex due to: {}", err.to_string())
+    })?;
+    if !regex.is_match(type_name.as_str()) {
+        return Err(bad_argument!(
+            "The type name {} is invalid, it should obey the regex {}",
+            type_name,
+            TYPE_NAME_REGEX
+        ));
+    }
+    let type_template = diesel::insert_into(jsonschema_types::table)
+        .values((
+            jsonschema_types::type_schema.eq(request.type_schema.clone()),
+            jsonschema_types::type_name.eq(request.type_name.clone()),
+            jsonschema_types::created_by.eq(user.email.clone()),
+        ))
+        .get_result::<JsonSchemaTypes>(&mut conn)
+        .map_err(|err| {
+            log::error!("failed to insert custom type with error: {}", err);
+            db_error!(err)
+        })?;
+    Ok(HttpResponse::Ok().json(type_template))
+}
+
+#[put("/{type_name}")]
+async fn update_type(
+    request: Json<Value>,
+    path: Path<String>,
+    db_conn: DbConnection,
+    user: User,
+) -> superposition::Result<HttpResponse> {
+    let DbConnection(mut conn) = db_conn;
+    let _ = JSONSchema::compile(&request).map_err(|err| {
+        log::error!(
+            "Invalid jsonschema sent in the request, schema: {:?} error: {}",
+            request,
+            err
+        );
+        bad_argument!(
+            "Invalid jsonschema sent in the request, validation error is: {}",
+            err.to_string()
+        )
+    })?;
+    let type_name = path.into_inner();
+    let regex = Regex::new(TYPE_NAME_REGEX).map_err(|err| {
+        unexpected_error!("could not parse regex due to: {}", err.to_string())
+    })?;
+    if !regex.is_match(type_name.as_str()) {
+        return Err(bad_argument!(
+            "The type name {} is invalid, it should obey the regex {}",
+            type_name,
+            TYPE_NAME_REGEX
+        ));
+    }
+    let timestamp = Utc::now().naive_utc();
+    let updated_type = diesel::update(jsonschema_types::table)
+        .filter(jsonschema_types::type_name.eq(type_name.clone()))
+        .set((
+            jsonschema_types::type_schema.eq(request.clone()),
+            jsonschema_types::created_by.eq(user.email),
+            jsonschema_types::last_modified.eq(timestamp),
+        ))
+        .get_result::<JsonSchemaTypes>(&mut conn)
+        .map_err(|err| {
+            log::error!("failed to insert custom type with error: {}", err);
+            db_error!(err)
+        })?;
+    Ok(HttpResponse::Ok().json(updated_type))
+}
+
+#[delete("/{type_name}")]
+async fn delete_type(
+    path: Path<String>,
+    db_conn: DbConnection,
+) -> superposition::Result<HttpResponse> {
+    let DbConnection(mut conn) = db_conn;
+    let type_name = path.into_inner();
+    let deleted_type =
+        diesel::delete(dsl::jsonschema_types.filter(dsl::type_name.eq(type_name)))
+            .get_result::<JsonSchemaTypes>(&mut conn)?;
+    Ok(HttpResponse::Ok().json(deleted_type))
```

Comment:
204

Reply:
204 is not appropriate here, there is a response body

Reply:
I don't think there is a need for the response body?

Reply:
The API caller gets information on what they deleted - frontends could reuse that to revert the action or log